src/userInterface/userInterface.ts

```diff
@@ -0,0 +1,502 @@
+          toChain,
+          networkHeightForUse,
+          toChain
+        );
+
+      const nativeRsnRatio: bigint = nativeTokenFeesInfo.rsnRatio;
+      const nativeRsnDivisor: bigint = nativeTokenFeesInfo.rsnRatioDivisor;
+      const nativeDecimals: number = nativeToken[toChain].decimals;
+      const assetRsnRatio: bigint = assetTokenFeesInfo.rsnRatio;
+      const assetRsnDivisor: bigint = assetTokenFeesInfo.rsnRatioDivisor;
+      const assetDecimals: number = token[toChain].decimals;
+      const networkFeeInAssetUnitToken: bigint =
+        this.calculateFeeToAssetUnitNetworkFee(
+          nativeRsnRatio,
+          nativeRsnDivisor,
+          nativeDecimals,
+          assetRsnRatio,
+          assetRsnDivisor,
+          assetDecimals,
+          baseNetworkFee
+        );
+
+      return networkFeeInAssetUnitToken;
+    } catch (error) {
+      throw new FeeConversionFailureException(
+        "Failed to convert fee to native network fee: " + error
+      );
+    }
+  }
+
+  // <utils>
+  public calculateFeeToAssetUnitNetworkFee(
+    nativeRsnRatio: bigint,
+    nativeRsnDivisor: bigint,
+    nativeDecimals: number,
+    assetRsnRatio: bigint,
+    assetRsnDivisor: bigint,
+    assetDecimals: number,
+    baseNetworkFee: bigint
+  ): bigint {
+    const nativePoweredDecimal: bigint = BigInt(Math.pow(10, nativeDecimals));
+    const assetPoweredDecimal: bigint = BigInt(Math.pow(10, assetDecimals));
+    const networkFeeInAssetUnitToken: bigint =
+      (baseNetworkFee *
+        assetPoweredDecimal *
+        nativeRsnRatio *
+        assetRsnDivisor) /
+      (assetRsnRatio * nativePoweredDecimal * nativeRsnDivisor);
+
+    return networkFeeInAssetUnitToken;
+  }
+
+  private checkTokenSupported(
+    fromChain: string,
+    tokenId: string,
+    toChain: string
+  ): string {
+    const tokens = this.tokenMap.search(fromChain, { tokenId: tokenId });
+    if (tokens.length <= 0) {
+      throw new TokenNotFoundException();
+    }
+
+    const token = tokens[0];
+
+    // Check if the chain exists
+    try {
+      this.tokenMap.getID(token, toChain);
+    } catch (error) {
+      throw new ChainNotSupportedException();
+    }
+
+    return this.tokenMap.getID(token, "ergo");
+  }
+
+  private async getNetworkHeight(
+    height: number,
+    chain: keyof typeof Networks
+  ): Promise<number> {
+    if (height === -1) {
+      return await this.network.GetHeight(chain);
+    } else {
+      return height;
+    }
+  }
+
+  private async getMinimumFeeBox(tokenId: string): Promise<MinimumFeeBox> {
+    const explorerUrl = this.network.GetExplorerUrl("ergo");
+
+    const minimumFee = new MinimumFeeBox(
+      tokenId,
+      this.config.FeeConfigTokenId,
+      this.ergoNetworkType,
+      explorerUrl,
+      this.logger
+    );
+
+    const fetchedBox = await minimumFee.fetchBox();
+    if (!fetchedBox) {
+      throw new FeeRetrievalFailureException("Fetch Minimum fee box failed");
```

Comment:
I suggest adding `tokenId` to Error for better debugging.

```suggestion
      throw new FeeRetrievalFailureException(`Failed to fetch Minimum fee box for token [${tokenId}]`);
```